consensus/state.go

```diff
@@ -1497,22 +1502,25 @@ func (cs *State) recordMetrics(height int64, block *types.Block) {
 				missingValidators++
 				missingValidatorsPower += val.VotingPower
 			}
+			if cs.privValidator != nil {
 
-			pv, err := cs.privValidator.GetPubKey()
-			if err != nil {
-				panic(err)
-			}
-
-			if cs.privValidator != nil && bytes.Equal(val.Address, pv.Address()) {
-				label := []string{
-					"validator_address", val.Address.String(),
+				pv, err := cs.privValidator.GetPubKey()
+				if err != nil {
+					panic(err)
```

Comment:
I think it's fine to log an error and return. If we don't log metrics for one/two blocks, nothing super bad will happen